core/src/main/java/com/linecorp/armeria/server/HttpServerPipelineConfigurator.java

```diff
@@ -93,22 +120,36 @@ protected void initChannel(Channel ch) throws Exception {
         final ChannelPipeline p = ch.pipeline();
         p.addLast(new FlushConsolidationHandler());
         p.addLast(ReadSuppressingHandler.INSTANCE);
+        configurePipeline(p, port.protocols(), null);
+    }
 
-        if (port.protocol().isTls()) {
-            assert sslContexts != null;
-            p.addLast(new SniHandler(sslContexts));
-            p.addLast(TrafficLoggingHandler.SERVER);
-            configureHttps(p);
-        } else {
-            p.addLast(TrafficLoggingHandler.SERVER);
-            configureHttp(p);
+    private void configurePipeline(ChannelPipeline p, Set<SessionProtocol> protocols,
+                                   @Nullable ProxiedAddresses proxiedAddresses) {
+        if (protocols.size() == 1) {
+            switch (protocols.iterator().next()) {
```

Comment:
Can use `Iterables.getFirst` which is a little more semantic.

Reply:
Fixed. Could have been better if there's a version that throws an exception when the iterable is empty.